sound/soc/sof/intel/hda-dsp.c

```diff
@@ -371,8 +371,13 @@ static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 	if (target_state->substate == SOF_HDA_DSP_D0I3) {
 		value = SOF_HDA_VS_D0I3C_I3;
 
-		/* disable DMA trace in D0I3 */
-		flags = HDA_PM_NO_DMA_TRACE;
+		/*
+		 * Trace DMA can be kept enabled when the DSP enters
+		 * D0I3 while the system is in S0. It must be disabled
+		 * when the system is suspending.
+		 */
+		flags = sdev->system_suspend_target == SOF_SUSPEND_NONE ?
+				0 : HDA_PM_NO_DMA_TRACE;
```

Comment:
@ranj063 Ack, this is exactly what I was meaning. This is actually even better as you get the info from sdev->system_suspend_target, so new info needs passing. Having many places in code passing "true/false" gets hard to read and fast. Thanks!